pkg/appctx/container.go

```diff
@@ -24,6 +27,18 @@ func (e ErrNoItemFound) Error() string {
 	return fmt.Sprintf("no item with key %v found", e.Key)
 }
 
+type KeyLock struct {
+	locks sync.Map
+}
+
+func (l *KeyLock) Lock(key any) *sync.Mutex {
+	lockItem, _ := l.locks.LoadOrStore(key, &sync.Mutex{})
```

Comment:
We shouldn't ignore errors, even if its very unlikely or impossible to get them. Let's just handle them to write clean code. :+1:

Reply:
LoadOrStore returns the item and a boolean flag if it was actually loaded (aka cash-hit).  I don't think we care at this point if the item was retrieved from the cache or not, but maybe if we also want to delete the mutex as you suggested after unlocking we could use this flag.